apm-sniffer/apm-sdk-plugin/thrift-plugin/src/main/java/org/apache/skywalking/apm/plugin/thrift/wrapper/ServerInProtocolWrapper.java

```diff
@@ -81,6 +85,24 @@ public TField readFieldBegin() throws TException {
             }
             return readFieldBegin();
         }
+        if (field.type == TType.STOP) {
+            Boolean haveCreatedSpan =
+                    (Boolean) ContextManager.getRuntimeContext().get(HAVE_CREATED_SPAN);
+            if (!haveCreatedSpan) {
```

Comment:
This is still an NPE risk. Take a look at my previous comment. If you are using `Boolean`, you need to `haveCreatedSpan != null &&` first.

If you are not familiar with auto-unboxing in Java, take a look at this, https://stackoverflow.com/questions/55100449/unexpected-nullpointerexception-while-autoboxing-unboxing-java-long-type-as-re

Reply:
I think I've initialize "haveCreatedSpan" in initial(), so it would't be NPE.
But It is necessary to add a precaution. Good advice.

Reply:
Yes, just be preciseness. :)